### ISSUE TITLE
MSSQL2 datasource option resolves as 'Other - JDBC driver (depreacted)' in Lucee

### ISSUE THREAD

**Issue by GaryStanton** [Ortus-Solutions/cfconfig#14]
The docs suggest MSSQL2 will resolve to the jTDS driver for MSSQL, but this doesn't seem to be the case. The resulting datasource has no connection string and verification results in the error:
`lucee.commons.lang.ClassException: cannot load class through its string name, because no definition for the class with the specified name [] could be found`

**Comment by bdw429s** [comment]
I think the docs you saw only apply to Adobe.  Does Lucee even have the jTDS driver for MSSQL by default?

**Comment by bdw429s** [comment]
Here is the applicable code: 
https://github.com/Ortus-Solutions/cfconfig/blob/master/models/BaseLucee.cfc#L959-L976

As you can see, it's not handling `MSSQL2`.  We can add that in, but I was thinking that wasn't really an out-of-the-box option for Lucee.

**Comment by GaryStanton** [comment]
Hi Brad,

I'm looking at the CFConfig docs on datasources here: https://cfconfig.ortusbooks.com/using-the-cli/command-overview/manage-datasources
From the wording there, I would expect that MSSQL resolves to the Microsoft driver and MSSQL2 resolves to the jTDS driver. 
Lucee has support for both out of the box:
![image](https://user-images.githubusercontent.com/1708375/41565981-398b3282-7351-11e8-86a8-398b5ac567b7.png)

**Comment by GaryStanton** [comment]
Hi Brad,

I'm looking at the CFConfig docs on datasources here: https://cfconfig.ortusbooks.com/using-the-cli/command-overview/manage-datasources
From the wording there, I would expect that MSSQL resolves to the Microsoft driver and MSSQL2 resolves to the jTDS driver. 
Lucee has support for both out of the box:
![image](https://user-images.githubusercontent.com/1708375/41565981-398b3282-7351-11e8-86a8-398b5ac567b7.png)

**Comment by bdw429s** [comment]
Makes sense.  Would you like to send a pull request to add this support? :)

**Comment by GaryStanton** [comment]
Sure I'd _like_ to... ;) 
I'm happy to give it a go, but I don't know how to fork and run a dev version of CFConfig... if you can give me some basic instructions I'll see what I can do.

**Comment by bdw429s** [comment]
You can run the tests by just doing this:
1. Fork and clone this repo locally
2. run `install` in the root
3. Run `server start` in the root
4. A browser window will open with the test results.

There is an installation of each supported CF engine under `/tests/resources/etc`. You might need to cd into each of those directories (they have a server.json) and start/stop each one for the Adobe tests to pass.  Or just ignore them since you really only care about Lucee.  I usually click a specific test case to run and then just do good old dump/abort in the files and debug it right from the browser.  You can run one of the Lucee tests for saving config to Lucee and then play around the JSON file it's using.

If you want to actually use CFConfig from the command line, also fork and clone the `commandbox-cfconfig` repo (but don't run `install` on it) and then run `package link` from the root of both repos and then `reload` your shell.  (uninstall any regular installation of CF Config first).  CFConfig won't show up in `list --system` now but it will be installed via symlinks and code changes will be picked up with a `reload` command.  This isn't a bad workflow, but it is a lot harder to debug stuff since you can't just use cfdump to see big structs, etc.

**Comment by GaryStanton** [comment]
Wow, thanks for all the detail! 
So I've managed to get a fork installed and the tests running. I can only see tests covering the reading and writing of the config files though, not for each _function_ of CFConfig. (e.g. saving a datasource)
I've made some changes to the code that I think will add support for MSSQL2, but I'm not sure how to test that? As in, I need to use this forked version of CFConfig to actually create a MSSQL2 datasource on the local server so that I can check it's working.

**Comment by GaryStanton** [comment]
Sorry Brad, I'm not very familiar with Commandbox, or indeed *box... 
I don't really understand how I can test my code by manually adding or reading stuff from the JSON files, I want to run the CLI command to see that it writes the correct string, and that it resolves correctly in the administrator.
I've tried to link the packages as you suggest, but I can't find the '/path/to/commandbox-cfconfig' .
I'm on Windows and I assume it's somewhere in `.CommandBox` inside my Users folder, but I've not had any luck finding it.

**Comment by bdw429s** [comment]
> I don't really understand how I can test my code by manually adding or reading stuff from the JSON files,

Here is a JSON file that I'm pretty sure is used in the tests.  These lines show the definition of a datasource to be imported into the CF engine.

https://github.com/Ortus-Solutions/cfconfig/blob/master/tests/resources/.CFConfig.json#L46-L60

There are tests that read from that JSON file:
https://github.com/Ortus-Solutions/cfconfig/blob/master/tests/specs/JSONConfigTest.cfc#L24-L29

and tests that write to the Lucee 4/5 servers

https://github.com/Ortus-Solutions/cfconfig/blob/master/tests/specs/Lucee5ServerConfigTest.cfc#L22-L26

You could create a test that moved config from the JSON file to a Lucee server home.  It would probably look like this:

```
// read in JSON
var JSONConfig = getInstance( 'JSONConfig@cfconfig-services' )
						.read( expandPath( '/tests/resources/.CFConfig.json' ) );

// Write it into a Lucee 5 server
var Lucee5ServerConfig = getInstance( 'Lucee5Server@cfconfig-services' )
					.setMemento( JSONConfig.getMemento() );
					.write( expandPath( '/tests/resources/lucee5/ServerHome/Lucee-Server' ) );
```

> I want to run the CLI command

Sure, you can do that.  It's really all the same code at the end of the day, whether the values came from the JSON file or from the CLI command.  Just link up the repos and use the CLI.  

> I can't find the /path/to/commandbox-cfconfig

I can't tell you where that is because it's wherever you put it :)  That's where you cloned your fork of the repo to.  This repo right here:
https://github.com/Ortus-Solutions/commandbox-cfconfig
You should have forked that and cloned it to some folder on your hard drive.  Whatever that folder is you cloned the repo to, that's what you use for the path.  

> I'm on Windows and I assume it's somewhere in .CommandBox inside my Users folder, but I've not had any luck finding it.

Nope, it's just the commandbox-cfonfig repo that I said to fork and clone along with the main cfconfig repo.

```
If you want to actually use CFConfig from the command line, also fork and clone the commandbox-cfconfig repo (but don't run install on it) and then run package link from the root of both repos and then reload your shell.
```

**Comment by GaryStanton** [comment]
Ok, I think I understand what you're saying with regard to the JSON, I'll see if I can figure that out.
Regarding the location of the repo, your instructions said that I'm supposed to link _both_ repos, so I assumed you meant the one installed to CommandBox and my fork... if not, I'm not sure where the second one comes from.
I've linked to the first one, my fork with the added code, but I'm not able to run any `CFConfig` commands from CommandBox.

I appreciate all the help... it would probably have taken you two minutes to just fix the issue yourself. ;)

**Comment by bdw429s** [comment]
>  I'm supposed to link both repos, so I assumed you meant the one installed to CommandBox and my fork.

No, CFConfig is actually a collection of packages that are assembled together.  When you install `commandbox-cfconfig` you are installing the CLI package, which links as a dependency, the `cfconfig-services` package (which is this repo here).  The services in turn rely on additional packages but we don't need to worry about them since I had you run `box install` to pull them in.  When I said to run link in both repos, I mean these two:
https://github.com/Ortus-Solutions/cfconfig
https://github.com/Ortus-Solutions/commandbox-cfconfig

**Comment by bdw429s** [comment]
> it would probably have taken you two minutes to just fix the issue yourself. ;)
Yeah, but then you wouldn't be learning all this stuff.  Teach a man to fish... :)
